### PR TITLE
Switch to justinmimb's types

### PR DESCRIPTION
**Pull request by miniBill**

### CODE REVIEW

File: src/Rfc3339.elm [hunk 1]
```diff
@@ -361,15 +334,15 @@ timeLocalParser =
...
                         case fracSeconds of
                             Nothing ->
-                                Parser.Advanced.succeed (toFloat second)
+                                Parser.Advanced.succeed ( second, 0 )
 
                             Just frac ->
-                                case String.toFloat (String.fromInt second ++ "." ++ frac) of
+                                case String.toInt (String.left 3 (frac ++ "000")) of
                                     Nothing ->
                                         Parser.Advanced.problem PrbExpectedAFloat
```
Comment (wolfadex) [inline src/Rfc3339.elm]: This problem either needs a rename or to be swapped out

File: tests/Test/Rfc3339.elm [hunk 2]
```diff
@@ -275,10 +256,10 @@ printing =
                     |> Expect.equal (Ok "00:00:00")
         , test "local time right after midnight" <|
             \() ->
-                "00:00:00.00001"
+                "00:00:00.001"
                     |> Rfc3339.parse
                     |> Result.map Rfc3339.toString
-                    |> Expect.equal (Ok "00:00:00.00001")
+                    |> Expect.equal (Ok "00:00:00.001")
```
Comment (wolfadex) [inline tests/Test/Rfc3339.elm:262]: Thinking we should add a test that acknowledges that we only support milliseconds. Where the time is something like `00:00:00.00001` and we get back 0 milliseconds

File: src/Rfc3339.elm [hunk 3]
```diff
@@ -237,51 +224,35 @@ dateTimeParser =
...
-                                daysInMonth date
-                        in
-                        if date.day > maxDays then
-                            Parser.Advanced.problem (PrbDayTooLarge maxDays)
 
-                        else
-                            Parser.Advanced.succeed dateTime
-            )
+checkDay : { a | year : Int, month : Time.Month, day : Int } -> Parser Date.Date
+checkDay date =
```
Comment (wolfadex) [inline src/Rfc3339.elm:166]: Nice!

File: tests/Test/Rfc3339.elm [hunk 4]
```diff
@@ -94,25 +97,15 @@ localDates =
                     |> Rfc3339.parse
                     |> Expect.equal
                         (Ok
-                            (Rfc3339.DateLocal
-                                { year = 1970
-                                , month = Time.Jan
-                                , day = 1
-                                }
-                            )
+                            (Rfc3339.DateLocal (Date.fromCalendarDate 1970 Time.Jan 1))
```
Comment (wolfadex) [inline tests/Test/Rfc3339.elm:100]: I wish `Date.fromCalendarDate` took a record 😞

File: tests/Test/Rfc3339.elm (hunk 2 above)
Comment (miniBill) [reply tests/Test/Rfc3339.elm:262]: As in - input with more precision, test that it comes out with less?

File: src/Rfc3339.elm (hunk 3 above)
Comment (miniBill) [reply src/Rfc3339.elm:166]: I know, right??? When I realized I was so happy with it

File: tests/Test/Rfc3339.elm (hunk 4 above)
Comment (miniBill) [reply tests/Test/Rfc3339.elm:100]: Eh...

File: tests/Test/Rfc3339.elm (hunk 2 above)
Comment (wolfadex) [reply tests/Test/Rfc3339.elm:262]: Exactly!

**Review by wolfadex — APPROVED** [review]

**Pull request merged**